woocommerce/woocommerce-ios
# [Better customer selection] Underline search term in customer cell fields

Closes: #10402 


## Description

With this PR we underline the substring of the customer search results (Name, email, username) that match the search query. 

## Changes
- Implement helper function to underline substrings on `NSMutableAttributedString`
- Rename components from `TitleAndSubtitleAndDetailTableViewCell` to `UnderlineableTitleAndSubtitleAndDetailTableViewCell` to make it more explicit
- Add underlined text property to view model
- Underline strings in `UnderlineableTitleAndSubtitleAndDetailTableViewCell`
- Keep the search term in `CustomerSearchUICommand`
- Wire the search term to the cell view models in the `CustomerSearchUICommand`

## Testing instructions


1. Go to Orders
2. Tap on + to create a new order
3. Tap on + Add Customer Details
4. Search a string
5. Note that the search string is underlined in the customer results


https://github.com/woocommerce/woocommerce-ios/assets/1864060/8bb8b9aa-8efa-42ff-9c6c-249533e8890c

## Screenshots

See video


---
